flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptivebatch/BlockingResultInfo.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.scheduler.adaptivebatch;
+
+import java.util.List;
+
+/** The blocking result info, which will be used to calculate the vertex parallelism. */
+public class BlockingResultInfo {
+
+    private final List<Long> blockingPartitionSizes;
+
+    private final boolean isBroadcast;
+
+    public BlockingResultInfo(List<Long> blockingPartitionSizes, boolean isBroadcast) {
```

Comment:
I prefer to introduce 2 factory methods to make the flag easy to understand, and then make this ctor private. Maybe `createFromBroadcastResult()` and create `fromNonBroadcastResult()`?